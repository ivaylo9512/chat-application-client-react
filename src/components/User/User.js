import React from 'react';
import { useHistory } from 'react-router-dom'
import { useDispatch } from 'react-redux';
import { setCurrentChat } from 'app/slices/chatsSlice';

const User = ({user}) => {
    const history = useHistory();
    const dispatch = useDispatch();

    const setChat = () => {
        if(user.chatWithUser){
            history.push('/chat');
            dispatch(setCurrentChat(user.chatWithUser));
        }
    }

    return (
        <div>
            <div className='image'>
                <img alt='profile' src={user.profilePicture}/>
            </div>
            <div className='info'>
                <b>{user.username}</b>
                <span></span>    
            </div>
            <button onClick={setChat}>
                {user.chatWithUser 
                    ? 'open'
                    : 'add'
                }
            </button>
        </div>
    )
}

export default User