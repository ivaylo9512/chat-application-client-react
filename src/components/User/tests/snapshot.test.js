import User from 'components/User/User';
import { shallow } from 'enzyme';
import * as Redux from 'react-redux';

describe("RegisterSnapshotTests", () => {
    let selectorSpy;

    beforeAll(() => {
        selectorSpy = jest.spyOn(Redux, 'useSelector');
        jest.spyOn(Redux, 'useDispatch').mockReturnValue(jest.fn());
    })
    
    const createWrapper = (state, user) => {
        selectorSpy.mockReturnValue(state);
        
        return shallow(
            <User user={user}/> 
        )
    }

    it('should render snapshot with chatWithUser', () => {
    const wrapper = createWrapper(undefined,  {firstName: 'First', lastName: 'Last', profilePicture: 'image.png', chatWithUser: true});

        expect(wrapper).toMatchSnapshot();
    })

    it('should render snapshot without request without chat', () => {
        const wrapper = createWrapper(undefined,  {firstName: 'First', lastName: 'Last', profilePicture: 'image.png', chatWithUser: false, requestState: 'send'});
    
        expect(wrapper).toMatchSnapshot();
    })

    it('should render snapshot with request', () => {
        const wrapper = createWrapper({ isLoading: false, state: 'send'},  {firstName: 'First', lastName: 'Last', profilePicture: 'image.png', chatWithUser: false, requestState: 'send'});
    
        expect(wrapper).toMatchSnapshot();
    })

    it('should render snapshot with loading request', () => {
        const wrapper = createWrapper({ isLoading: true},  {firstName: 'First', lastName: 'Last', profilePicture: 'image.png', chatWithUser: false, requestState: 'send'});
    
        expect(wrapper).toMatchSnapshot();
    })
})