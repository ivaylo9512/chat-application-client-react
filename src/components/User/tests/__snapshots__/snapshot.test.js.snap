// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User snapshot tests should render snapshot 1`] = `
<styled.div>
  <styled.div>
    <Styled(styled.button)>
      <img
        src="http://localhost:8080/images/download/undefined"
      />
    </Styled(styled.button)>
    <styled.div
      isInfoVisible={false}
    >
      <span>
        <b>
          First Last
        </b>
      </span>
      <styled.div
        isInfoVisible={false}
      >
        <span>
          <b>
            First Last
          </b>
        </span>
        <span>
          <b />
        </span>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.button
        data-testid="toggleInfo"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                192,
                512,
                Array [],
                "f129",
                "M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v212.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 24 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z",
              ],
              "iconName": "info",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </styled.button>
      <Memo()
        chatWithUser={
          Object {
            "id": 5,
          }
        }
        initialMessage="send"
        requestId={4}
      />
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`User snapshot tests should render snapshot with request error 1`] = `
<styled.div>
  <styled.div>
    <Styled(styled.button)>
      <img
        src="http://localhost:8080/images/download/undefined"
      />
    </Styled(styled.button)>
    <styled.div
      isInfoVisible={false}
    >
      <span>
        <b>
          First Last
        </b>
      </span>
      <styled.div
        isInfoVisible={false}
      >
        <span>
          <b>
            First Last
          </b>
        </span>
        <span>
          <b />
        </span>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.button
        data-testid="toggleInfo"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                192,
                512,
                Array [],
                "f129",
                "M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v212.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 24 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z",
              ],
              "iconName": "info",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </styled.button>
      <Memo()
        chatWithUser={
          Object {
            "id": 5,
          }
        }
        initialMessage="send"
        requestId={4}
      />
    </styled.div>
  </styled.div>
  <styled.div>
    Request not found.
  </styled.div>
</styled.div>
`;
